src/sentry/tasks/on_demand_metrics.py

```diff
@@ -268,17 +268,25 @@ def _set_widget_on_demand_state(
     is_low_cardinality: bool | None,
     enabled_features: Set[str],
 ):
-    extraction_state = _determine_extraction_state(specs, is_low_cardinality, enabled_features)
+    extraction_state = _determine_extraction_state(
+        widget_query, specs, is_low_cardinality, enabled_features
+    )
     spec_hashes = [hashed_spec[0] for hashed_spec in specs]
 
-    DashboardWidgetQueryOnDemand.objects.update_or_create(
+    on_demand = DashboardWidgetQueryOnDemand.objects.get_or_create(
         dashboard_widget_query=widget_query,
         defaults={
             "spec_hashes": spec_hashes,
             "extraction_state": extraction_state,
         },
     )
 
+    if on_demand.can_extraction_be_auto_overriden():
+        on_demand.extraction_state = extraction_state
+
+    on_demand.spec_hashes = spec_hashes
+    on_demand.save()
```

Comment:
`update_fields` wouldn't apply here since every time I would call it would be when most fields need to be updated, plus this is a very small table with very few columns. Switched this to `.asave()` though, til.